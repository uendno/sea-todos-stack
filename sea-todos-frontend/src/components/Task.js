import React from 'react';
import PropTypes from 'prop-types';

export default function Task({task, toggleTask}) {
  const {text, completed} = task;
  return (
    <li className={completed ? 'completed' : ''} onClick={() => toggleTask(task)}>
      <div className="view">
        <input
          className="toggle"
          type="checkbox"
          onChange={() => {}}
          onClick={() => toggleTask(task)}
          checked={completed}
        />
        <label>{text}</label>
      </div>
    </li>
  )
}

Task.propTypes = {
  task: PropTypes.object.isRequired,
  toggleTask: PropTypes.func.isRequired,
}
