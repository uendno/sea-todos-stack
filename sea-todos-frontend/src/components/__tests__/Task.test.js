import React from 'react';
import { shallow, mount } from 'enzyme';
import Task from '../Task';

describe('components/Task', () => {
  let wrapper;
  let props;

  const setup = () => {
    wrapper = shallow(<Task {...props}/>);
  }

  beforeEach(() => {
    props = {
      task: {
        text: 'Some task',
        completed: true
      },
      toggleTask: jest.fn()
    }
  })

  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  })
});
