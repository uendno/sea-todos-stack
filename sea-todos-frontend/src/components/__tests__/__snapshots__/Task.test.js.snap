// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Task should render correctly 1`] = `
<li
  className="completed"
>
  <div
    className="view"
  >
    <input
      checked={true}
      className="toggle"
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <label>
      Some task
    </label>
  </div>
</li>
`;
